robot/lib/Src/drivers/MCP23017.cpp

```diff
@@ -24,14 +20,13 @@ void MCP23017::reset() {
 }
 
 void MCP23017::writeRegister(MCP23017::Register regAddress, uint16_t data) {
-    char buffer[] = {regAddress, (char)(data & 0xff), (char)(data >> 8)};
-    _i2c.write(_i2cAddress, buffer, sizeof(buffer));
+    std::vector<uint8_t> buffer{static_cast<uint8_t>(data & 0xff),
+                                static_cast<uint8_t>(data >> 8)};
+    _i2c->transmit(_i2cAddress, regAddress, buffer);
 }
 
 uint16_t MCP23017::readRegister(MCP23017::Register regAddress) {
-    char buffer[2];
-    _i2c.write(_i2cAddress, (char*)&regAddress);
-    _i2c.read(_i2cAddress, buffer, 2);
+    std::vector<uint8_t> buffer = _i2c->receive(_i2cAddress, regAddress, 2);
```

Comment:
Putting it here because I don't have a better place for it (probably in the mtrain repo but I'm lazy).

`<kyle dynamic allocation rant>`

I am not a fan of this. This heap-allocates a 2-byte value. Repeatedly. To read a register. AND the size is bounded at compile time (I assume) because you are passing in 2.

I would _strongly_ prefer to at least have a second API for receive that allows you to pass in a pointer to a buffer and a size. Same with transmit above. This is overkill and physically pains me.

`</kyle dynamic allocation rant>`